src/backend/distributed/commands/type.c

```diff
@@ -209,20 +210,8 @@ PlanAlterTypeStmt(AlterTableStmt *stmt, const char *queryString)
 
 	Assert(stmt->relkind == OBJECT_TYPE);
 
-	if (creating_extension)
-	{
-		/*
-		 * extensions should be created separately on the workers, types cascading from an
-		 * extension should therefor not be propagated.
-		 */
-		return NIL;
-	}
-
-	/*
-	 * Only distributed types should be propagated
-	 */
 	typeAddress = GetObjectAddressFromParseTree((Node *) stmt, false);
-	if (!IsObjectDistributed(typeAddress))
+	if (!ShouldPropagateAlterType(typeAddress))
```

Comment:
Should we use this flag on `ReplicateAllDependenciesToNode()` and `create_distributed_function()`? I wonder if we can find a point to add this check such that any future object also not propagated?

Reply:
> I wonder if we can find a point to add this check such that any future object also not propagated?

I have been thinking about this, and is not feasible in the current form of our code. `EnableDDLPropagation` did this somewhat right for its purpose as we putt all calls for `Plan*` and `Process*` functions in a block guarded by that flag.

This will not work for new objects being propagated as they are often routed through overloaded statements by Postgres, eg. `ALTER TYPE ... ADD ATTRIBUTE ...` is parsed into a `AlterTableStmt` struct which is reused for tables and types, and possibly more. We don't want to put the whole handling of `AlterTableStmt` in a block guarded by the `EnableObjectPropagation` flag as that would intervene with changes on distributed tables. For now we need to be diligent and keep adding this check to the objects we want to exclude from distribution such as the upcoming functions.

> Should we use this flag on ReplicateAllDependenciesToNode() and create_distributed_function()?

`ReplicateAllDependenciesToNode()` is a complicated one. Currently it will behave very strange when there are objects in `pg_dist_object`. It will most likely fail to order the dependencies already recorded before the flag was turned off. A solution to this might be to filter `pg_dist_object` by with `SupportedDependenciesByCitus` as that will fail all but the schemas. Those are a requirement to still create on the new node and be able to distribute tables to the node.

Since `create_distributed_function` is a deliberate call by the user, I wonder if we should execute, error or potentially warn when object propagation is turned off. I think executing without a warning is potentially dangerous, the warning should be something like:
```
distributing functions when object propagation is turned off will prevent citus from keeping the function on the coordinator and the workers in sync
```

Completely erroring on the UDF feels funky. I don't think we give any warning or error on `create_distributed_table` when ddl propagation is turned off, so maybe executing without any warning is a sane thing to do.

Reply:
> distributing functions when object propagation is turned off will prevent citus from keeping the function on the coordinator and the workers in sync

I think this sounds good.